src/ast/FuncDecl.ts

```diff
@@ -9,26 +9,93 @@ import {Tokenizer} from "../util/Tokenizer";
  * e.g. function modifier (async | static)? str
  *          params []
  *          comments?
- *          (generate getter)?
- *          (generate setter)?
  *          (return TYPE)?
  */
 export default class FuncDecl extends AstNode {
 
     modifier: string;
-    isAsync: boolean;
-    isStatic: boolean;
+    isAsync: boolean = false;
+    isStatic: boolean = false;
     name: string;
     params: VarList;
     comment: CommentDecl;
-    generateGetter: boolean;
-    generateSetter: boolean;
     returnType: string;
 
     public parse(context: Tokenizer): any {
```

Comment:
I want to refactor this, since there's just so much going on, but for a first pass...meh

Reply:
This guy's a big boye, maybe we need more ASTNodes.

Reply:
@scveloso done

Reply:
thank